src/compiler/program.ts

```diff
@@ -738,6 +738,10 @@ namespace ts {
             getDiagnosticsProducingTypeChecker,
             getCommonSourceDirectory,
             emit,
+            getDeclarationEmitPath: fileName => {
+                const file = getSourceFile(fileName);

```

Comment:
why does this need to be on the program? `getDeclarationEmitOutputFilePath` is already available?

Reply:
`getEmitHost()` isn't though. That function appears to need `host.getCompilerOptions`, `host.getCurrentDirectory`, `host.getCommonSourceDirectory`, and `host.getCanonicalFileName`. `getCurrentDirectory` and `getCanonicalFileName` can be answered without a `Program`, but I don't think the other two can.

Reply:
getCurrentDirectory and getCanonicalFileName should be there on project and can be answered from there.

Reply:
That still leaves `getCompilerOptions` and `getCommonSourceDirectory` unfortunately.

Reply:
getCompilerOptions  is also present on project and getCommonSourceDirectory can be added as a private method to the project that redirects to the program ?